File: CodingSamples/Java/Language/ObjectClassTest/Interval.java
//a class that does not explicitly extend any other class implicitly
//extends java.lang.Object which is the root of class inheritance tree in Java
class Interval {

	private int min;
	private int sec;

	public Interval(int m, int s) {
		min = m + s / 60;
		sec = s % 60;
	}

	public int minutes() {
		return min;
	}

	public int seconds() {
		return sec;
	}

	public int time() {
		return 60 * min + sec;
	}

	public String toString() {
		if(sec < 10)
			return min + ":0" + sec;
		return min + ":" + sec;
	}

	public int hashCode() {
		return min + sec;
	}

	public boolean equals(Object other) {
		if(other instanceof Interval that){
			return (min == that.min) && (sec == that.sec);
		}
		return false;
	}
}

File: CodingSamples/DotNet/Runtime/GenTypeTest/DemoApp/Interval.cs
class Interval
{
    //read-only property can only be set in a constructor
    public int Minutes { get; }

    public int Seconds { get; }

    public Interval(int min, int sec)
    {
        Minutes = min + sec / 60;
        Seconds = sec % 60;
    }

    public int Time()
    {
        return 60 * Minutes + Seconds;
    }

    public override string ToString()
    {
        if(Seconds < 10)
            return Minutes + ":0" + Seconds;
        return Minutes + ":" + Seconds;
    }

    public static Interval operator+(Interval lhs, Interval rhs)
    {
        return new Interval(lhs.Minutes + rhs.Minutes, lhs.Seconds + rhs.Seconds);
    }

    public override int GetHashCode()
    {
        return Minutes + Seconds;
    }

    public override bool Equals(object obj)
    {
        if(obj is Interval that)
        {
            return (Minutes == that.Minutes) && (Seconds == that.Seconds);
        }
        return false;
    }
}